ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -495,6 +496,34 @@ type BiosConfigRef struct {
 // IronicInspect
 type IronicInspect string
 
+type SiteConfigMap struct {
+	Name      string            `yaml:"name"`
+	Namespace string            `yaml:"namespace"`
+	Data      map[string]string `yaml:"data"`
+}
+
+// Provide custom YAML unmarshal for SiteConfigMap which provides default values
+func (rv *SiteConfigMap) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type ValueDefaulted SiteConfigMap
+	var defaults = ValueDefaulted{
+		Name:      "ztp-site-configmap",
+		Namespace: "ztp-site",
+	}
+
+	out := defaults
+	err := unmarshal(&out)
```

Comment:
This is more readable and follows Go's error handling:

*rv = SiteConfigMap(out)
err := unmarshal(&out)
if err != nil {
		return err
	}
return nil

Reply:
I agree, but I wanted to keep it consistent with the rest of the file.